vortex-file/src/read/mod.rs

```diff
@@ -59,6 +64,15 @@ pub enum BatchRead {
     Batch(ArrayData),
 }
 
+#[derive(Debug)]
+pub enum MetadataRead {
+    /// Layout has no metadata
+    None,
+    /// Additional IO is required
+    ReadMore(Vec<MessageLocator>),
+    Batches(Vec<ArrayData>),
```

Comment:
What about this? Then Columnar can return `MetadataRead<Vec<Option<ArrayData>>>` and Chunked can return `MetadataRead<Option<ArrayData>>`

```suggestion
pub enum MetadataRead<T> {
    /// Additional IO is required
    ReadMore(Vec<MessageLocator>),
    Batches(T),
```

Reply:
Figuring out this API is the one missing piece, I don't think this will work with the LayoutReader trait.

Reply:
Ok I think I have something that works, LMK what you think.